proposal (copy).md

```diff
@@ -0,0 +1,139 @@
+   high-level overview. Relevant experts in that code-section should be asked
+   to provide their review.
+3. Follow up on issues and pull requests that are going stale. Pandas enjoys a
+   large number of one-time pull requests from a member of the community. These
+   contributors are often unable to respond quickly to multiple rounds of
+   review. This can result in the contribution going stale and the Pull Request
+   abandoned.
+
+### Native String Data Type
+
+Currently, pandas stores text data in an ``object`` -dtype NumPy array.
+Each array stores Python strings. While pragmatic, since we rely on NumPy
+for storage and Python for string operations, this is memory inefficient
+and slow. We'd like to provide a native string type for pandas.
+
+The most obvious alternative is [Apache Arrow][arrow]. Arrow's data model is
+well-suited to storing arrays of textual data. Arrow arrays can already be
+stored inside a DataFrame using the [extension array interface][ea interface].
+We propose the following
+
+1. Work with the Arrow developers to implement the [string operations][string
+   ops] on Arrow memory. This may be done in the Arrow C++ library (benefiting
+   all libraries using it, including Python, R, and Ruby), or in a
+   Python-specific manor using Numba (@lam2015numba) (following the
+   [Fletcher][fletcher] proof of concept).
+2. Provide a backwards-compatible transition for current pandas users. This
+   large of a change (including a new dependency) may require a period where
+   users must opt into the new behavior.
+
+### Documentation Validation
+
+To improve the quality and consistency of Pandas documentation, we've developed
+tooling to check docstrings in a variety of ways. Every docstring is checked for
+
+1. Consistency: Ensuring that every docstring has the same components
+   (parameters, return values, notes, examples, etc.). Ensuring that each
+   section is formatted correctly.
+2. Completeness: Ensuring that every function parameter is documented.
+3. Correctness: Ensuring that the examples run correctly.
+
+When a user submits a Pull Request to pandas, their changes to the documentation
+are automatically checked, and informative error messages inform them of any
+issues.
+
+Like many other projects, pandas uses the
+[numpydoc](https://numpydoc.readthedocs.io/en/latest/) standard for writing
+docstrings. With the collaboration of the numpydoc maintainers, we'd like to
+move the project-agnostic tooling we've written to a different project that any
+numpydoc-using project can benefit from.
+
+If possible, we'd like this project to be undertaken by a member of an
+unrepresented minority, with mentorship provided by the pandas maintaiers. This
+project primarily requires experience with *using* pandas, NumPy, and related
+libraries, rather that deep knowledge of pandas' internals. Pandas has a diverse
+community, just not at the maintainer level (yet).
```

Comment:
Should we also include the improvement of the tools itself (I think there are still quite some open issues in pandas to improve the docstring validation) here?

Reply:
Done in #4.